src/arfsdao.ts

```diff
@@ -611,58 +660,115 @@ export class ArFSDAO extends ArFSDAOAnonymous {
-						break;
-					case 'Unix-Time':
-						drive.unixTime = +value;
-						break;
-					default:
-						break;
-				}
+		const drive = new ArFSPrivateDriveBuilder(driveId, this.arweave, driveKey);
+		return await drive.build();
+	}
+
+	async getPrivateFolder(folderId: FolderID, driveKey: DriveKey): Promise<ArFSPrivateFolder> {
+		const folderBuilder = new ArFSPrivateFolderBuilder(folderId, this.arweave, driveKey);
+		return await folderBuilder.build();
+	}
+
+	async getAllFoldersOfPrivateDrive(
+		driveId: DriveID,
+		driveKey: DriveKey,
+		latestRevisionsOnly = false
+	): Promise<ArFSPrivateFolder[]> {
+		let cursor = '';
+		let hasNextPage = true;
+		const allFolders: ArFSPrivateFolder[] = [];
+
+		while (hasNextPage) {
+			const gqlQuery = buildQuery(
+				[
+					{ name: 'Drive-Id', value: driveId },
+					{ name: 'Entity-Type', value: 'folder' }
+				],
+				cursor
+			);
+
+			const response = await this.arweave.api.post(graphQLURL, gqlQuery);
+			const { data } = response.data;
+			const { transactions } = data;
+			const { edges } = transactions;
+			hasNextPage = transactions.pageInfo.hasNextPage;
+
+			const folders: Promise<ArFSPrivateFolder>[] = edges.map(async (edge: GQLEdgeInterface) => {
+				cursor = edge.cursor;
+				const { node } = edge;
+				const folderBuilder = await ArFSPrivateFolderBuilder.fromArweaveNode(node, this.arweave, driveKey);
+				return await folderBuilder.build(node);
 			});
+			allFolders.push(...(await Promise.all(folders)));
+		}
+		return latestRevisionsOnly ? allFolders.filter(latestRevisionFilter) : allFolders;
+	}
+
+	async getPrivateFilesWithParentFolderIds(
+		folderIDs: FolderID[],
+		driveKey: DriveKey,
+		latestRevisionsOnly = false
+	): Promise<ArFSPrivateFile[]> {
+		let cursor = '';
+		let hasNextPage = true;
+		const allFiles: ArFSPrivateFile[] = [];
+
+		while (hasNextPage) {
+			const gqlQuery = buildQuery(
+				[
+					{ name: 'Parent-Folder-Id', value: folderIDs },
+					{ name: 'Entity-Type', value: 'file' }
+				],
+				cursor
+			);
 
-			// Get the drives transaction ID
-			drive.txId = node.id;
+			const response = await this.arweave.api.post(graphQLURL, gqlQuery);
+			const { data } = response.data;
+			const { transactions } = data;
+			const { edges } = transactions;
+			hasNextPage = transactions.pageInfo.hasNextPage;
+			const files: Promise<ArFSPrivateFile>[] = edges.map(async (edge: GQLEdgeInterface) => {
+				const { node } = edge;
+				cursor = edge.cursor;
+				const fileBuilder = await ArFSPrivateFileBuilder.fromArweaveNode(node, this.arweave, driveKey);
+				return await fileBuilder.build(node);
+			});
+			allFiles.push(...(await Promise.all(files)));
 		}
-		return drive.build(this.wallet as JWKWallet, drivePassword, this.arweave);
+		return latestRevisionsOnly ? allFiles.filter(latestRevisionFilter) : allFiles;
+	}
+
+	/**
+	 * Lists the children and self of certain private folder
+	 * @param {FolderID} folderId the folder ID to list children of
```

Comment:
Docblock needs updates

Reply:
614fe0b796d24b994e18927fb971ae683c9f8ccf

Reply:
And driveKey